File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.js
function solution(s) {
    const n = s.length;
    let answer = n;
    
    for (unit_num = 1; unit_num <= Math.floor(n/2); unit_num++) {
        let result = "";
        let prev_str = s.slice(0, unit_num);
        let cnt = 1;
        
        for (i = unit_num; i <= n; i += unit_num) {
            let now_str = s.slice(i, i+unit_num);
            
            if (now_str === prev_str) {
                cnt += 1;
            } else {
                result += (cnt > 1 ? String(cnt) : "") + prev_str;
                prev_str = now_str;
                cnt = 1;
            }
        }
        
        result += (cnt > 1 ? String(cnt)  : "") + prev_str;
        
        answer = Math.min(answer, result.length);
    }
    
    return answer;
}

File: 프로그래머스/2/60057. 문자열 압축/문자열 압축.py
def solution(s):
    n = len(s)
    answer = n
    
    for unit_num in range(1, n//2+1):
        result = ""
        prev_str = s[0:unit_num]
        cnt = 1
        
        for i in range(unit_num, n, unit_num):
            now_str = s[i:i+unit_num]
            if prev_str == now_str:
                cnt += 1
            else:
                if cnt == 1:
                    result += prev_str
                else:
                    result += (str(cnt) + prev_str)
                prev_str = now_str
                cnt = 1

        if cnt == 1:
            result += prev_str
        else:
            result += (str(cnt) + prev_str)
        
        answer = min(answer, len(result))
                    
    return answer